### ISSUE TITLE
False Positive of E0308 due to undetected Clone impl. 

### ISSUE THREAD

**Issue by C0D3-M4513R** [intellij-rust/intellij-rust#10811]
## Environment

* **IntelliJ Rust plugin version:** 0.4.200.5421-232
* **Rust toolchain version:** 1.70.0 (90c541806 2023-05-31) x86_64-pc-windows-gnu
* **IDE name and version:** CLion 2023.2 (CL-232.8660.186)
* **Operating system:** Windows 10.0
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr

## Steps to reproduce

```rust
#[derive(Clone)]
struct Test<A>{
    a: std::sync::Arc<A>
}

impl<A> Test<A>{
    pub fn new(a: A) -> Self{
        Self{
            a: std::sync::Arc::new(a)
        }
    }
}

fn main() {
    let test = Test::new(String::new());
    a(&test);
    println!("Success!");
}

fn a<A>(a:&Test<A>) -> Test<A> {
    a.clone() //mismatched types [E0308] expected `Test<A>`, found `&Test<A>` 
}
```
